code/game/machinery/camera/camera.dm

```diff
@@ -166,6 +169,15 @@
 					M.reset_perspective(null)
 					to_chat(M, "The screen bursts into static.")
 
+/obj/machinery/camera/emag_act(mob/user) //DONKSTATION CHANGE: Adds emag effects to Security Cameras. 8a15645
```

Comment:
this emag act should add emag flag to the obj
not sure but it might be done in parent proc (which you should call!)